src/Message.chpl

```diff
@@ -365,11 +371,11 @@ module Message {
     */
     proc parseMessageArgs(json_str: string, size: int) throws {
         var pArr = jsonToPdArray(json_str, size);
-        var paramArr: [0..#size] ParameterObj;
-        forall (i, j_str) in zip(0..#size, pArr) {
-            paramArr[i] = parseParameter(j_str);
+        var param_list: list(ParameterObj) = new list(ParameterObj);
+        for (i, j_str) in zip(0..#size, pArr) {
+            param_list.append(parseParameter(j_str));
```

Comment:
the `i` is no longer used and (provided order of the `ParameterObj`s within `param_list` doesn't matter) i think we should be able to still do this in parallel
```chapel
var param_list = new list(ParameterObj, parSafe=true);
forall j_str in pArr with (ref param_list) {
  param_list.append(parseParameter(j_str));
}
```

Reply:
I intentionally ignored the HDF5 Legacy code that is not in the command map. That code should be removed fully after the next release cycle.

I am going to test the parallel structure with gasnet and 2 locals, but that should work.

Reply:
Updated to use parallel loop